### DISCUSSION TITLE
.properties file and cucumber report

### DISCUSSION THREAD

**Discussion by aydinserbest** [cucumber/common#2179] [General]
To generate a Cucumber report with JUnit 5:

1-When I add @ConfigurationParameter(key = "cucumber.publish.enabled", value = "true"), the report is generated.
2-When I remove this annotation and add a cucumber.properties file in the src/test/resources directory with the content cucumber.publish.enabled=true, the report is not generated.
3-However, when I rename the .properties file to junit-platform.properties, the report is generated.
Questions:
1-How does JUnit 5 read the .properties file without any additional code or configuration?
2-How can the effect of renaming the file be explained, i.e., why does it recognize the file immediately when it is named junit-platform.properties?

**Comment by mpkorstanje** [comment]
I think you may have misunderstood something in the architecture.

As noted in  Cucumber [JUnit Platform Engine - Configuration Options](https://github.com/cucumber/cucumber-jvm/tree/main/cucumber-junit-platform-engine#configuration-options):

> Cucumber receives its configuration from the JUnit Platform. To see how these can be supplied; see the JUnit documentation [4.5. Configuration Parameters](https://junit.org/junit5/docs/current/user-guide/#running-tests-config-params). For documentation on Cucumber properties, see [Constants](https://github.com/cucumber/cucumber-jvm/blob/main/cucumber-junit-platform-engine/src/main/java/io/cucumber/junit/platform/engine/Constants.java).

When using the the JUnit Platform Suite Engine you are using:

```
Your IDE / Maven / Gradle -> JUnit Platform -> Suite Engine -> Junit Platform -> Cucumber Engine -> Cucumber
````

So when Cucumber looks up properties, as provided by the JUnit Platform it will look at the following sources in order:

1. Properties from `@ConfigurationParameter` annotations
2. Properties from Your IDE / Maven / Gradle
3. Properties from the JVM
4. Properties from `junit-platform.properties`

The JUnit Platform knows to look at `junit-platform.properties` because it is the provider of all properties. The JUnit Platform also does not know or care about `cucumber.properties`. Only when renaming it to `junit-platform.properties` the file is read.

Once https://github.com/junit-team/junit5/pull/3345 lands it will be possible to use `@ConfigurationParametersResource("cucumber.properties")` in a suite. At that point `cucumber.properties` would also be read by the JUnit Platform.

**Reply by aydinserbest** [reply]
![Screenshot 2024-07-06 at 14 27 08](https://github.com/cucumber/common/assets/101752975/2cfc861d-1ba7-4c0e-9047-01010901c652)
The name issue has been resolved, thank you. Additionally, I will review the document.
